src/state/lottery/helpers.ts

```diff
@@ -0,0 +1,295 @@
+      lastTicketId,
+      amountCollectedInCake,
+      finalNumber,
+      cakePerBracket,
+      countWinnersPerBracket,
+      rewardsBreakdown,
+    } = lotteryData
+
+    const statusKey = Object.keys(LotteryStatus)[status]
+    const serializedCakePerBracket = cakePerBracket.map((cakeInBracket) => ethersToSerializedBigNumber(cakeInBracket))
+    const serializedCountWinnersPerBracket = countWinnersPerBracket.map((winnersInBracket) =>
+      ethersToSerializedBigNumber(winnersInBracket),
+    )
+    const serializedRewardsBreakdown = rewardsBreakdown.map((reward) => ethersToSerializedBigNumber(reward))
+
+    return {
+      isLoading: false,
+      status: LotteryStatus[statusKey],
+      startTime: startTime?.toString(),
+      endTime: endTime?.toString(),
+      priceTicketInCake: ethersToSerializedBigNumber(priceTicketInCake),
+      discountDivisor: discountDivisor?.toString(),
+      treasuryFee: treasuryFee?.toString(),
+      firstTicketId: firstTicketId?.toString(),
+      lastTicketId: lastTicketId?.toString(),
+      amountCollectedInCake: ethersToSerializedBigNumber(amountCollectedInCake),
+      finalNumber,
+      cakePerBracket: serializedCakePerBracket,
+      countWinnersPerBracket: serializedCountWinnersPerBracket,
+      rewardsBreakdown: serializedRewardsBreakdown,
+    }
+  } catch (error) {
+    return {
+      isLoading: true,
+      status: LotteryStatus.PENDING,
+      startTime: '',
+      endTime: '',
+      priceTicketInCake: '',
+      discountDivisor: '',
+      treasuryFee: '',
+      firstTicketId: '',
+      lastTicketId: '',
+      amountCollectedInCake: '',
+      finalNumber: null,
+      cakePerBracket: [],
+      countWinnersPerBracket: [],
+      rewardsBreakdown: [],
+    }
+  }
+}
+
+export const fetchCurrentLotteryIdAndMaxBuy = async () => {
+  try {
+    const calls = ['currentLotteryId', 'maxNumberTicketsPerBuyOrClaim'].map((method) => ({
+      address: getLotteryV2Address(),
+      name: method,
+    }))
+    const [[currentLotteryId], [maxNumberTicketsPerBuyOrClaim]] = (await multicallv2(
+      lotteryV2Abi,
+      calls,
+    )) as ethers.BigNumber[][]
+
+    return {
+      currentLotteryId: currentLotteryId ? currentLotteryId.toString() : null,
+      maxNumberTicketsPerBuyOrClaim: maxNumberTicketsPerBuyOrClaim ? maxNumberTicketsPerBuyOrClaim.toString() : null,
+    }
+  } catch (error) {
+    return {
+      currentLotteryId: null,
+      maxNumberTicketsPerBuyOrClaim: null,
+    }
+  }
+}
+
+export const processRawTicketsResponse = (ticketsResponse: UserTicketsResponse): LotteryTicket[] => {
+  const ticketIds = ticketsResponse[0]
+  const ticketNumbers = ticketsResponse[1]
+  const ticketStatuses = ticketsResponse[2]
+
+  if (ticketIds.length > 0) {
+    return ticketIds.map((ticketId, index) => {
+      return {
+        id: ticketId.toString(),
+        number: ticketNumbers[index].toString(),
+        status: ticketStatuses[index],
+      }
+    })
+  }
+  return []
```

Comment:
I don't think you need this clause. `map` will return an empty array.

Reply:
Good point. 

I think it'll crash if it calls `map` on `undefined`/`null`... but it'll crash on the `if` anyway. I've tested this pretty extensively but just in case I'll improve the `if` to `if (ticketIds?.length > 0) {`. 

The `return []` then becomes a catch for when `ticketsResponse` is undefined, rather than an empty array.